packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -170,17 +170,15 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     childManager.didStartLayout();
     childManager.setDidUnderflow(false);
 
-    final double itemExtent = this.itemExtent;
-
     final double scrollOffset = constraints.scrollOffset + constraints.cacheOrigin;
     assert(scrollOffset >= 0.0);
     final double remainingExtent = constraints.remainingCacheExtent;
     assert(remainingExtent >= 0.0);
     final double targetEndScrollOffset = scrollOffset + remainingExtent;
 
     final BoxConstraints childConstraints = constraints.asBoxConstraints(
-      minExtent: itemExtent,
-      maxExtent: itemExtent,
+      minExtent: itemExtent?? 0,
+      maxExtent: itemExtent?? computeMaxScrollOffset(constraints, itemExtent),
```

Comment:
This seems incorrect. The child can't be between 0 and maxScrollOfsset. It has to be whatever you said in `itemExtents` for this index.

Reply:
Yes, this is a hack to just demo the idea. I agree that there are many places like this that are very fragile or plainly wrong, but I think it's sufficient to demonstrate the idea how `itemExtents` could potentially speed things up. I'll further clarify it in the title and PR description.